homeassistant/components/sensor/min_max.py

```diff
@@ -98,7 +100,7 @@ def async_min_max_sensor_state_listener(entity, old_state, new_state):
             if self._unit_of_measurement != new_state.attributes.get(
                     ATTR_UNIT_OF_MEASUREMENT):
                 _LOGGER.warning("Units of measurement do not match")
-                return
```

Comment:
Update: when reloading configuration, unit_of_measurement may change - either the one configured with this sensor or those configured with the calculated entities. So the following situation can happen with the old code (correct me, if i'm wrong)
- first incoming unit is wrong and remembered by this sensor. when it's fixed in config and config is reloaded, it won't work

Suggestions to solve this:
- ignore units, use only the unig configured for min_max
- remember units for all values. if all matches, use this unit. otherwise return ERROR as unit, that's clearly visible in the UI